packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts

```diff
@@ -948,6 +1060,120 @@ declare module 'lightning/analyticsWaveApi' {
+        outputRows?: number;
+    }
+
+    /**
+     * Replicates data from an external source object into a Wave as a dataset. Replicated Datasets are not intended to be visualized directly, but are used like a cache to speed up other workflows which refer to the same source object.
+     *
+     * https://developer.salesforce.com/docs/atlas.en-us.bi_dev_guide_rest.meta/bi_dev_guide_rest/bi_resources_replicateddatasets.htm#ReplicatedDatasetInputRepresentation
+     *
+     * Keys:
+     *    (none)
+     */
+    export interface ReplicatedDatasetInputRepresentation {
+        advancedProperties?: Array<{
+            [key: string]: any;
+        }>;
+        connectionMode?: string;
+        connectorId?: string;
+        fullRefresh?: boolean;
+        passThroughFilter?: string;
+        rowLevelSharing?: boolean;
+        sourceObjectName?: string;
+    }
+
+    /**
+     * Replicates data from an external source object into a Wave as a dataset. Replicated Datasets are not intended to be visualized directly, but are used like a cache to speed up other workflows which refer to the same source object.
+     *
+     * https://developer.salesforce.com/docs/atlas.en-us.bi_dev_guide_rest.meta/bi_dev_guide_rest/bi_resources_replicateddatasets.htm#ReplicatedDatasetRepresentation
+     *
+     * Keys:
+     *    id (string): uuid
+     */
+    export interface ReplicatedDatasetRepresentation {
+        assetSharingUrl?: string | null;
+        createdBy?: WaveUserRepresentation;
+        createdDate?: string;
+        description?: string;
+        id: string;
+        label?: string;
+        lastAccessedDate?: string | null;
+        lastModifiedBy?: WaveUserRepresentation | null;
+        lastModifiedDate?: string | null;
+        name?: string;
+        namespace?: string;
+        permissions?: PermissionsRepresentation | null;
+        type: string;
+        url: string;
+        advancedProperties: Array<AdvancedPropertyValueRepresentation>;
+        connectionMode?: string;
+        connector: DataConnectorRepresentation;
+        datasetId?: string;
+        fieldCount?: number;
+        fieldsUrl: string;
+        filterApplied: boolean;
+        lastRefreshedDate?: string;
+        outputSource?: OutputSourceRepresentation;
+        passThroughFilter?: string;
+        replicationDataflowId?: string;
+        rowLevelSharing?: boolean;
+        sourceObjectName: string;
+        status?: string;
+        supportedConnectionModes?: Array<string>;
+        uuid: string;
+    }
+
+    /**
+     * A list of configuration metadata that specifies how to replicate each field of a Replicated Dataset.
+     *
+     * https://developer.salesforce.com/docs/atlas.en-us.bi_dev_guide_rest.meta/bi_dev_guide_rest/bi_resources_replicateddatasets_id_fields.htm#ReplicatedFieldCollectionInputRepresentation
+     *
+     * Keys:
+     *    (none)
+     */
+    export interface ReplicatedFieldCollectionInputRepresentation {
+        fields: Array<{}>;
```

Comment:
This is the generated type and the raml doesn't list any properties for the objects that can get added to `fields`, but it doesn't seem right. The doc links says `fields` is a `Replicated​Field​Input​Representation[]`, but `Replicated​Field​Input​Representation` isn't specified in the raml.
Maybe just switch it to `fields: Array<{ [key: string]: any }>` or similar?

Reply:
Yeah, it the current implementation looks weird,  the inheritance is`ReplicatedFieldCollectionInputRepresentation -> ReplicatedFieldInputListRepresentation -> List<ReplicatedFieldInputRepresentation>. Also it seems for output representation, the generated raml cannot recognize the type on list property, i found similar issue on `WaveCollectionInputRepresentation.pins`. We do have `ReplicatedFieldInputRepresentation` in the generated raml on core side but it is never added to LDS raml.

Reply:
After discussing this a bit, it's fine to leave this as-is for now since it doesn't appear to break anything, and there's some issues in the underlying types that we would need to address to get the right generated type showing up and we can do that later.